manager/controlapi/ca_rotation_test.go

```diff
@@ -0,0 +1,771 @@
+	require.NoError(t, err)
+	defer differentExtServer.Stop()
+	require.NoError(t, differentExtServer.EnableCASigning())
+
+	testcases = []*rootCARotationTestCase{
+		{
+			description: "different cert than current results in new root rotation",
+			rootCA:      initialLocalRootCA,
+			caConfig: api.CAConfig{
+				SigningCACert: differentInitialCert,
+				SigningCAKey:  initialLocalRootCA.CAKey,
+				ForceRotate:   5,
+			},
+			expectRootCA:           getRootCAWithRotation(getExpectedRootCA(true), differentInitialCert, initialLocalRootCA.CAKey, nil),
+			expectedGeneratedCross: true,
+		},
+		{
+			description: "different cert than current, external->internal, results in new root rotation",
+			rootCA:      initialExternalRootCA,
+			caConfig: api.CAConfig{
+				SigningCACert: differentInitialCert,
+				SigningCAKey:  initialLocalRootCA.CAKey,
+				ForceRotate:   5,
+				ExternalCAs: []*api.ExternalCA{
+					{
+						URL:    initExtServer.URL,
+						CACert: initialExternalRootCA.CACert,
+					},
+				},
+			},
+			expectRootCA:           getRootCAWithRotation(getExpectedRootCA(false), differentInitialCert, initialLocalRootCA.CAKey, nil),
+			expectedGeneratedCross: true,
+		},
+		{
+			description: "different cert than current, internal->external, results in new root rotation",
+			rootCA:      initialLocalRootCA,
+			caConfig: api.CAConfig{
+				SigningCACert: differentInitialCert,
+				ForceRotate:   5,
+				ExternalCAs: []*api.ExternalCA{
+					{
+						// we need a different external server, because otherwise the external server's cert will fail to validate
+						// (not signed by the right cert - note that there's a bug in go 1.7 where this is not needed, because the
+						// subject names of cert names aren't checked, but go 1.8 fixes this.)
+						URL:    differentExtServer.URL,
+						CACert: differentInitialCert,
+					},
+				},
+			},
+			expectRootCA:           getRootCAWithRotation(getExpectedRootCA(true), differentInitialCert, nil, nil),
+			expectedGeneratedCross: true,
+		},
+	}
+	runValidTestCases(t, testcases, &localRootCA)
+
+	// These will change the root rotation because the desired cert are different than the current root rotation cert and hash
+	for i, testcase := range testcases {
+		switch i {
+		case 0:
+			testcase.rootCA = getRootCAWithRotation(testcase.rootCA, rotationCert, rotationKey, crossSigned)
+			testcase.description = "different cert and key than root rotation contents results in replacing root rotation"
+
+		case 1:
+			testcase.rootCA = getRootCAWithRotation(testcase.rootCA, rotationCert, nil, crossSigned)
+			testcase.description = "different cert than root rotation contents, even if external->internal, results in replacing root rotation"
+		case 2:
+			testcase.rootCA = getRootCAWithRotation(testcase.rootCA, rotationCert, rotationKey, crossSigned)
+			testcases[2].description = "different cert than root rotation contents, even if internal->external, results in replacing root rotation"
+		}
+	}
```

Comment:
I was trying to preserve the original testcases' configurations and expected root CA values.  It was to test that whatever the initial `RootRotation` value was, because the cert is different, it will just be replaced completely by a new root rotation with `differentInitialCert`, and that any external CAs required by the old root rotation will no longer be required.

I think this can probably be replaced by a single case.